src/simple-keyring.ts

```diff
@@ -0,0 +1,226 @@
+    const privateKey = wallet.privateKey.toString('hex');
+    return decrypt({ privateKey, encryptedData });
+  }
+
+  // personal_signTypedData, signs data along with the schema
+  async signTypedData(
+    address: Hex,
+    typedData: any,
+    opts: Record<string, string> = { version: SignTypedDataVersion.V1 },
+  ) {
+    // Treat invalid versions as "V1"
+    const version = Object.keys(SignTypedDataVersion).includes(
+      opts.version || '',
+    )
+      ? (opts.version as SignTypedDataVersion)
+      : SignTypedDataVersion.V1;
+
+    const privateKey = this.#getPrivateKeyFor(address, opts);
+    return signTypedData({ privateKey, data: typedData, version });
+  }
+
+  // get public key for nacl
+  async getEncryptionPublicKey(withAccount: Hex, opts: Record<string, string>) {
+    const privKey = this.#getPrivateKeyFor(withAccount, opts);
+    const publicKey = getEncryptionPublicKey(privKey.toString('hex'));
+    return publicKey;
+  }
+
+  #getPrivateKeyFor(
+    address: Hex,
+    options: Record<string, unknown> = { withAppKeyOrigin: '' },
+  ) {
+    if (!address) {
+      throw new Error('Must specify address.');
+    }
+    const wallet = this.#getWalletForAccount(address, options);
+    return wallet.privateKey;
+  }
+
+  // returns an address specific to an app
+  async getAppKeyAddress(address: Hex, origin: string) {
+    if (!origin || typeof origin !== 'string') {
+      throw new Error(`'origin' must be a non-empty string`);
+    }
+    const wallet = this.#getWalletForAccount(address, {
+      withAppKeyOrigin: origin,
+    });
+    const appKeyAddress = add0x(bufferToHex(publicToAddress(wallet.publicKey)));
+    return appKeyAddress;
+  }
+
+  // exportAccount should return a hex-encoded private key:
+  async exportAccount(address: Hex, opts = { withAppKeyOrigin: '' }) {
+    const wallet = this.#getWalletForAccount(address, opts);
+    return wallet.privateKey.toString('hex');
+  }
+
+  removeAccount(address: string) {
+    if (
+      !this.#wallets
+        .map(({ publicKey }) =>
+          bufferToHex(publicToAddress(publicKey)).toLowerCase(),
+        )
+        .includes(address.toLowerCase())
+    ) {
+      throw new Error(`Address ${address} not found in this keyring`);
+    }
+
+    this.#wallets = this.#wallets.filter(
+      ({ publicKey }) =>
+        bufferToHex(publicToAddress(publicKey)).toLowerCase() !==
+        address.toLowerCase(),
+    );
+  }
+
+  #getWalletForAccount(
+    account: string | number,
+    opts: Record<string, unknown> = {},
+  ) {
+    const address = normalize(account);
+    let wallet = this.#wallets.find(
+      ({ publicKey }) => bufferToHex(publicToAddress(publicKey)) === address,
+    );
+    if (!wallet) {
+      throw new Error('Simple Keyring - Unable to find matching address.');
+    }
+
+    if (opts.withAppKeyOrigin) {
+      const { privateKey } = wallet;
+      const appKeyOriginBuffer = Buffer.from(
+        opts.withAppKeyOrigin as string,
```

Comment:
Any way to avoid this typecast? This would allow for `opts.withAppKeyOrigin` to be something we don't expect, which seems undesirable.